src/node/interfaces.cpp

```diff
@@ -935,6 +936,27 @@ class MinerImpl : public Mining
         return BlockRef{tip->GetBlockHash(), tip->nHeight};
     }
 
+    BlockRef waitTipChanged(uint256 current_tip, MillisecondsDouble timeout) override
+    {
+        // Interrupt check interval
+        const MillisecondsDouble tick{1000};
+        auto now{std::chrono::steady_clock::now()};
+        auto deadline = now + timeout;
+        // std::chrono does not check against overflow
+        if (deadline < now) deadline = std::chrono::steady_clock::time_point::max();
```

Comment:
It was in response to: https://github.com/bitcoin/bitcoin/pull/30409#discussion_r1713879417

Reply:
Nice catch. I believe this can be simplified to:


```diff
--- a/src/node/interfaces.cpp
+++ b/src/node/interfaces.cpp
@@ -941,9 +941,7 @@ public:
         // Interrupt check interval
         const MillisecondsDouble tick{1000};
         auto now{std::chrono::steady_clock::now()};
-        auto deadline = now + timeout;
-        // std::chrono does not check against overflow
-        if (deadline < now) deadline = std::chrono::steady_clock::time_point::max();
+        const auto deadline{now + timeout};
         {
             WAIT_LOCK(notifications().m_tip_block_mutex, lock);
             while ((notifications().m_tip_block == uint256() || notifications().m_tip_block == current_tip) && !chainman().m_interrupt) {
```

This previous code written to handle the case where `now + timeout` calculation would overflow, basically assuming both values were unsigned integer values and overflow behavior would be well defined. But since `timeout` is a floating point number where "overflow" would just result in `inf` being set, trying to deal with it this way doesn't make sense.